CHANGELOG.md

```diff
@@ -7,12 +7,17 @@ All notable changes to this project will be documented in this file.
 ### Added
 
 - Added `Option::as_ref_or_else` to `utils` ([#717]).
+- Add `iter()` and `into_iter()` methods to `KeyValuePairs<T>` ([#720]).
+- Delegate `iter()` and `into_iter()` for `Labels` and `Annotations` ([#720]).
 
 ### Changed
 
 - Split `utils` into submodules ([#717]).
+- Bump rust to 1.75.0 ([#720]).
+- Replace `BTreeSet` with `Vec` for underlying `KeyValuePairs` collection ([#720]).
```

Comment:
Not looked into the code, but this smells like another https://github.com/stackabletech/airflow-operator/pull/372, WDYT @adwk67?

Reply:
The airflow issue was related to what common_config.env_overrides returns, so it was a similar principle but slightly different.

Reply:
Happy to revert that commit.
Including @Techassi since he was privy to conversation about it.

Reply:
Oh yeah that might have been the reason why I used `BTreeSet` in the first place. Sorry about that!

Let's maybe add a doc comment for this on `KeyValuePairs`, so that we don't stumble over this again in the future.

Reply:
- Reverted in 
38191d8 and 7a8a421
- documented in a1902d6:
  ![image](https://github.com/stackabletech/operator-rs/assets/10092581/ecb88ea7-300e-42a2-83c5-4b3dda15d4ac)